src/backup_restore.cpp

```diff
@@ -9,24 +9,43 @@
 
 namespace vpd
 {
+BackupAndRestoreStatus BackupAndRestore::m_backupAndRestoreStatus =
+    BackupAndRestoreStatus::NotStarted;
+
 BackupAndRestore::BackupAndRestore(const nlohmann::json& i_sysCfgJsonObj) :
     m_sysCfgJsonObj(i_sysCfgJsonObj)
 {
-    m_backupAndRestoreCfgJsonObj = jsonUtility::getParsedJson(
-        i_sysCfgJsonObj.value("backupRestoreConfigPath", ""));
+    std::string l_backupAndRestoreCfgFilePath =
+        i_sysCfgJsonObj.value("backupRestoreConfigPath", "");
+    try
```

Comment:
if empty , would a throw be better saying that the file is missing path or path empty in file

Reply:
so if you feel you can debug in this format then I am good